test/Container/RoleServiceFactoryTest.php

```diff
@@ -44,10 +44,10 @@ public function testCanCreateRoleService(): void
             ],
         ]);
 
-        $container = $this->getMockBuilder(ContainerInterface::class)->getMock();
-
-        $container->expects($this->at(0))->method('get')->with(ModuleOptions::class)->willReturn($options);
-        $container->expects($this->at(1))->method('get')->with(RoleProviderPluginManager::class)->willReturn(new RoleProviderPluginManager($this->getMockBuilder(ContainerInterface::class)->getMock()));
+        $container = new ServiceManager(['services' => [
```

Comment:
if we go psr-11, let's remove service-manager in tests, too and use prophecy for mocking the interface

Reply:
this PR is not about removing service manager but it brings an ability to use something else if needed. If we would decide to remove it we can do it in separate PR I guess.

Reply:
Is this article also true for phpunit? http://everzet.com/post/72910908762/conceptual-difference-between-mockery-and-prophecy. I'm only familiar with phpunit, perhaps I should broaden my horizon

Reply:
@prolic  I can mock this if you think this is really needed